features/crc/soft/FastCRC.h

```diff
@@ -0,0 +1,202 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_FAST_CRC_H
+#define MBED_FAST_CRC_H
+
+#include "mbed_crc.h"
+#include "mbed.h"
+
+template <typename T>
+class FastCRC : public CRC
```

Comment:
It may be worth documenting this template parameter. I'm not entirely sure what value it adds to the class (the internal table uses T as elements? Does the resulting CRC size depend on T or the polynomial argument?)

Reply:
Resulting CRC size will always be uint32_t  (masked based on CRC width). Kept it generic in CRC class

Reply:
Does this need to be a template then? you can extend an instantiated template:
``` cpp
class FastCRC : public CRC<uint32_t>
```

Reply:
CRC class is generic and will be used for hardware CRC as well. If you check hardware CRC implementation of target devices, they all return 32-bit CRC (masked) based on polynomial. Could be because of C implementation, but in order to have consistent behavior it would be good to not have template in base class.

Any inputs from HAL team?